File: LeetCode/Array/Golang/Leetcode1403-minimum-subsequence-in-non-increasing-order.go
package sbgo

import "sort"

func minSubsequence(nums []int) []int {
    ans,acc,sum,n := []int{},0,0,len(nums)
    for _,item := range nums {
        sum += item
    }
    sort.Ints(nums)
    for i := n - 1;i >= 0 && acc <= sum;i-- {
        ans = append(ans,nums[i])
        acc += nums[i]
        sum -= nums[i]
    } 
    return ans
}

File: LeetCode/Array/JS/Leetcode1403-minimum-subsequence-in-non-increasing-order.js
var minSubsequence = function(nums) {
    const n = nums.length;
    nums.sort((a,b)=> a - b);
    let [acc,sum,ans] = [0,nums.reduce((item,x)=>x + item,0),[]];
    for(let i = n - 1;i >= 0 && acc <= sum;i--) {
        acc += nums[i];
        sum -= nums[i];
        ans.push(nums[i]);
    }
    return ans;
};
